ts/features/mvl/screens/__test__/MvlRouterScreen.test.tsx

```diff
@@ -0,0 +1,168 @@
+        const globalState = appReducer(
+          undefined,
+          applicationChangeState("active")
+        );
+        const store = createStore(appReducer, globalState as any);
+        const render = renderComponent(store);
+
+        render.store.dispatch(mvlDetailsLoad.success(mvlMockData));
+
+        expect(
+          render.component.queryByTestId("MvlDetailsScreen")
+        ).not.toBeNull();
+      });
+    });
+  });
+  describe("When the screen is rendered with a failure in feature.mvl.byId", () => {
+    it("Should render the MvlLoadingScreen", () => {
+      const globalState = appReducer(
+        undefined,
+        applicationChangeState("active")
+      );
+      const store = createStore(appReducer, globalState as any);
+      store.dispatch(
+        mvlDetailsLoad.failure({
+          kind: "generic",
+          id: mvlMockId,
+          value: new Error("An error")
+        })
+      );
+      const render = renderComponent(store);
+
+      expect(render.component.queryByTestId("MvlLoadingScreen")).not.toBeNull();
+    });
+
+    it("Should dispatch a loading request", () => {
+      const globalState = appReducer(
+        undefined,
+        applicationChangeState("active")
+      );
+      const store = createStore(appReducer, globalState as any);
+      store.dispatch(
+        mvlDetailsLoad.failure({
+          kind: "generic",
+          id: mvlMockId,
+          value: new Error("An error")
+        })
+      );
+      const mockStore = configureMockStore<GlobalState>();
+      const finalStore: ReturnType<typeof mockStore> = mockStore({
+        ...store.getState()
+      } as GlobalState);
+      renderComponent(finalStore);
+
+      expect(finalStore.getActions()).toStrictEqual([
+        mvlDetailsLoad.request(mvlMockId)
+      ]);
+    });
+  });
+
+  describe("When the screen is rendered with a success in feature.mvl.byId", () => {
+    it("Should render the MvlDetailsScreen", () => {
+      const globalState = appReducer(
+        undefined,
+        applicationChangeState("active")
+      );
+      const store = createStore(appReducer, globalState as any);
+      store.dispatch(mvlDetailsLoad.success(mvlMockData));
+      const render = renderComponent(store);
+
+      expect(render.component.queryByTestId("MvlDetailsScreen")).not.toBeNull();
+    });
+
+    it("Should NOT dispatch a loading request", () => {
+      const globalState = appReducer(
+        undefined,
+        applicationChangeState("active")
+      );
+      const store = createStore(appReducer, globalState as any);
+      store.dispatch(mvlDetailsLoad.success(mvlMockData));
+      const mockStore = configureMockStore<GlobalState>();
+      const finalStore: ReturnType<typeof mockStore> = mockStore({
+        ...store.getState()
+      } as GlobalState);
+      renderComponent(finalStore);
+
+      expect(finalStore.getActions()).toStrictEqual([]);
+    });
+  });
+});
+
+const renderComponent = (store: Store) => ({
```

Comment:
There is some boilerplate that might be better in this helper for maintainability. For instance, it may take a list of actions to be dispatched (order must be guaranteed).

The goal is to move from [the code above](https://github.com/pagopa/io-app/pull/3545/files?authenticity_token=5xv%2FKnlMs07vVb%2FzTwZsJSSwNHj8Ig%2ByNsItZe9dtLyRSCK%2BLGXNsJ%2BUk61ew%2Fg7RqrtCFOVhWkUGDQWja%2B0oA%3D%3D&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx&file-filters%5B%5D=No+extension&hide-deleted-files=true#diff-8db75839990ef4df2c08f44c6ade0f1f620392415f8277803b85fbeeddfa4ab9R131):

```typescript
const globalState = appReducer(
        undefined,
        applicationChangeState("active")
      );
      const store = createStore(appReducer, globalState as any);
      store.dispatch(mvlDetailsLoad.success(mvlMockData));
      const render = renderComponent(store);
```

To this:

```typescript
renderComponent([mvlDetailsLoad.success(mvlMockData)])
```

Reply:
Better for sure! Refactored in 03bf2d5. It's possible to have a sequence of action that will be dispatched before the component rendering.